hashicorp/consul-ui-toolkit
# feat: add Copy Block to toolkit

### :hammer_and_wrench: Description

**Add capability to allow actionable metadata attributes (text) to have a specific hover style and click action.** 

- For text within the template that is actionable (like a piece of text or link that needs to be copied):

-- The list item must be static (_in control by toolkit user_)
-- The text that is actionable when hovered will show a change in background (surface/interactive-hover) and has the potential to have an icon added to the right (in many cases will be a copy icon)
-- When the text and or Icon is clicked it should perform the action (in the above use-case, copy the text highlighted when hovered)
```
<L.Content><Cut::CopyBlock @clipboardText="Service b">Service b</Cut::CopyBlock></L.Content>
```

- For text on hover that show a tooltip (they should not get this hover style) (_managed with hds-tooltip_)
```
  <Cut::ListItem as |L|>
    <L.Content><div {{hds-tooltip 'hello'}}>Service a</div></L.Content>
  </Cut::ListItem>
```

**NOTE**: Pr updates design-system-components version and adapts SideNav

### :camera_flash: Screenshots

https://github.com/hashicorp/consul-ui-toolkit/assets/10027860/cb53543d-d224-4f7e-8e51-a2ba1a3926c8

https://github.com/hashicorp/consul-ui-toolkit/assets/10027860/27ce7ed0-a254-43bc-8d2b-1a8d99802143

### :link: External Links


[CC-4893](https://hashicorp.atlassian.net/browse/CC-4893)
[Figma](https://www.figma.com/file/SenC8kTQNUJKCvOvcL8WPu/Consul-Component-Library?type=design&node-id=642-89768&t=TBA4QXbceXhDST4E-0)
### :building_construction: How to Build and Test the Change


1. go to vercel root + /list-item
2. Check out **Static** block
<img width="801" alt="image" src="https://github.com/hashicorp/consul-ui-toolkit/assets/10027860/1f5777f6-b807-4e0c-a62d-a78c6e7e892b">

### :+1: Definition of Done

- [x] New functionality works?

### :speech_balloon: Using the [Netlify feedback ladder](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify/)

- :mount_fuji: **[mountain]**: Blocking and requires immediate action.
- :moyai: **[boulder]**: Blocking.
- :white_circle: **[pebble]**: Non-blocking, but requires future action.
- :hourglass_flowing_sand: **[sand]**: Non-blocking, but requires future consideration.
- :sparkles: **[dust]**: Non-blocking. "Take it or leave it"


[CC-4893]: https://hashicorp.atlassian.net/browse/CC-4893?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
